src/views/Home.vue

```diff
@@ -1,18 +1,142 @@
 <template>
   <div class="home">
-    <img alt="Vue logo" src="../assets/logo.png">
-    <HelloWorld msg="Welcome to Your Vue.js + TypeScript App"/>
+    <div class="input">
+      <InputField
+        type="text"
+        placeholder="First Name"
+        name="firstName"
+        :invalid-input="submitted && v$.user.firstName.$invalid"
+        :error-message="`${$t('validationError.firstName')}`"
+        @changeValue="updateUser"
+      />
+    </div>
+    <div class="input">
+      <InputField
+        type="text"
+        placeholder="Last Name"
+        name="lastName"
+        :invalid-input="submitted && v$.user.lastName.$invalid"
+        :error-message="`${$t('validationError.lastName')}`"
+        @changeValue="updateUser"
+      />
+    </div>
+    <div class="input">
+      <InputField
+        type="email"
+        placeholder="Email"
+        name="email"
+        :invalid-input="submitted && v$.user.email.$invalid"
+        :error-message="`${$t('validationError.email')}`"
+        @changeValue="updateUser"
+      />
+    </div>
+    <div class="input">
+      <InputField
+        type="password"
+        placeholder="Password"
+        name="password"
+        :invalid-input="submitted && v$.user.password.$invalid"
+        :error-message="`${$t('validationError.password')}`"
+        @changeValue="updateUser"
+      />
+    </div>
+    <div class="input">
+      <InputField
+        type="password"
+        placeholder="Confirm Password"
+        name="confirmPassword"
+        :invalid-input="submitted && v$.user.confirmPassword.$invalid"
+        :error-message="`${$t('validationError.confirmPassword')}`"
+        @changeValue="updateUser"
+      />
+    </div>
+    <Button :message="`${$t('button.register')}`" @onClick="registerUser" />
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue';
-import HelloWorld from '@/components/HelloWorld.vue'; // @ is an alias to /src
+import { defineComponent, ref } from "vue";
+import InputField from "@/components/InputField.vue";
+import Button from "@/components/Button.vue";
+import useVuelidate from "@vuelidate/core";
+import { required, email, minLength, sameAs } from "@vuelidate/validators";
+import { User } from "@/types/interface";
 
 export default defineComponent({
-  name: 'Home',
+  name: "Home",
   components: {
-    HelloWorld,
+    InputField,
+    Button,
+  },
+  setup() {
+    const user = ref<User>({
+      firstName: "",
+      lastName: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+    });
+    const submitted = ref<boolean>(false);
```

Comment:
I would like to see us build a paradigm where each view and component maintains 5 states - DEFAULT, LOADING, LOADED, ERROR, LOADED_NO_DATA

DEFAULT - the default state when its rendered.
LOADING - When an async operation is being performed and we are waiting for result
LOADED - Operation completed, with some data in response
ERROR - Operation completed, but failed
LOADED_NO_DATA - Operation completed, but no data

Maintaining explicit state for a component helps to make code readable and easy to argue. It should be abstracted well so even Button component can use it